### ISSUE TITLE
undefined method `close' for nil:NilClass

### ISSUE THREAD

**Issue by bcg62** [sous-chefs/line#60]
Seeing this on latest cookbook version.

```
           System Info:
           ------------
           chef_version=13.2.20
           platform=centos
           platform_version=6.8
           ruby=ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]
           program_name=chef-solo worker: ppid=4737;start=13:16:55;
           executable=/opt/chef/bin/chef-solo
```

```
             ================================================================================
             Error executing action `edit` on resource 'delete_lines[remove test from /tmp/test]'
             ================================================================================

             NoMethodError
             -------------
             undefined method `close' for nil:NilClass

             Cookbook Trace:
             ---------------
             /tmp/kitchen/cache/cookbooks/line/resources/delete_lines.rb:41:in `ensure in block in class_from_file'
             /tmp/kitchen/cache/cookbooks/line/resources/delete_lines.rb:42:in `block in class_from_file'
```

**Comment by damacus** [comment]
Thanks for the bug report @bcg62 what's the code you're running to get that?

**Comment by MarkGibbons** [comment]
Hi Dan,
  I can reproduce the error by deleting a line from a file that doesn't exist.  I'll put together something in the  rescue to not close the file if we raise an error for a non existent file.  Should have it by Monday.

I'll check the other actions for the same type of handling.
Mark

**Comment by damacus** [comment]
@markgibbons that was my gut instinct. 

Thanks! 

We should probably pop in a test for this too

**Comment by MarkGibbons** [comment]
The test is in the PR, it's not a great test but it covers the basic idea.

**Comment by damacus** [comment]
Annoyingly we want a failing test... But obviously that's hard to get a positive back from. 

This method must fail in this particular way. 

We need a positive and negative test suite here, to make sure we don't break the "working" one.

**Comment by MarkGibbons** [comment]
I can do that.  Later in the week.

**Comment by MarkGibbons** [comment]
Added chef spec tests to verify the failing mode creates the proper message.

**Comment by damacus** [comment]
Closed by #61
